src/Playwright/Transport/Channels/ElementHandleChannel.cs

```diff
@@ -49,7 +50,25 @@ internal override void OnMessage(string method, JsonElement? serverParams)
         switch (method)
         {
             case "previewUpdated":
-                PreviewUpdated?.Invoke(this, serverParams.Value.GetProperty("preview").ToString());
+                var preview = string.Empty;
+                try
+                {
+                    preview = serverParams.Value.GetProperty("preview").ToString();
+                }
+                catch
+                {
+                    var matchResult = Regex.Match(serverParams.Value.ToString(), "\"preview\":\"(.*)\"");
```

Comment:
I think there should be a better way to accomplish this. What's the exception you're getting?

Reply:
I went ahead here: https://github.com/microsoft/playwright/pull/28038

Investigation: we trimmed wrongly, it ended up in an invalid utf8 char and the .NET JSON parser didn't like this.